src/utils/cognito.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  AdminConfirmSignUpCommand,
+  AttributeType,
+  AuthenticationResultType,
+  CognitoIdentityProviderClientConfig,
+  InitiateAuthCommand,
+  SignUpCommand,
+} from '@aws-sdk/client-cognito-identity-provider';
+import { getCognitoClient } from './internal';
+
+/**
+ * Get credentials for an existing user.
+ */
+export const cognitoSignIn = async (params: {
+  clientId: string;
+  username: string;
+  password: string;
+  config?: CognitoIdentityProviderClientConfig;
+}): Promise<AuthenticationResultType> => {
+  const { clientId, username, password, config } = params;
+  const client = getCognitoClient(config);
+  const { AuthenticationResult } = await client.send(
+    new InitiateAuthCommand({
+      AuthFlow: 'USER_PASSWORD_AUTH',
+      ClientId: clientId,
+      AuthParameters: {
+        USERNAME: username,
+        PASSWORD: password,
+      },
+    }),
+  );
+  if (!AuthenticationResult) {
+    throw new Error('AuthenticationResult is undefined');
+  }
+  return AuthenticationResult;
+};
+
+/**
+ * Create a new user in cognito, auto confirm it and return its credentials.
+ */
+export const cognitoSignUp = async (params: {
+  clientId: string;
+  userPoolId: string;
+  username: string;
+  password: string;
+  attributes?: AttributeType[];
+  config?: CognitoIdentityProviderClientConfig;
+}): Promise<AuthenticationResultType> => {
+  const { clientId, userPoolId, username, password, attributes, config } =
+    params;
+  const client = getCognitoClient(config);
+
+  await client.send(
+    new SignUpCommand({
+      ClientId: clientId,
+      Username: username,
+      Password: password,
+      UserAttributes: attributes,
+    }),
```

Comment:
Here too, we might want to use `AdminCreateUserCommand`.
The issue with classical signup is that it might require email/phone verification, etc.

Reply:
OK

There is a `temporaryPassWord` [option](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminCreateUser.html#API_AdminCreateUser_RequestSyntax)

Reply:
> I will double-check that maybe there's an admin method that is able to create an account with a permanent password (not a temporary one).

Found this https://stackoverflow.com/a/56948249/7549756, it's available in AWS SDK V3 [AdminSetUserPasswordCommand](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-cognito-identity-provider/classes/adminsetuserpasswordcommand.html) 🎉

Reply:
Yes! sorry, that's what I wanted to point out. 
I sent the wrong info :)

Reply:
> OK
> 
> There is a `temporaryPassWord` [option](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminCreateUser.html#API_AdminCreateUser_RequestSyntax)

This is an optional parameter for setting a `temporaryPassword`, if it's not passed Amazon Cognito will generate a random temporary password for us.

Reply:
Done switched to use admin methods like  `AdminCreateUserCommand` and `AdminSetUserPasswordCommand`, https://github.com/serverless-guru/sls-jest/pull/73/commits/0aec2ce8d3bd9c42e733d06965353769c2f75e0f